### PR TITLE
Applying to matplotlib for GSoC 2017

### PR DESCRIPTION
**Pull request by vidursatija**
Checklist for prospect students for Google Summer of Code 2017:

- [x] I contacted NumFOCUS mentors before writing my proposal
- [x] I showed you my contribution (it can be any form: proof-of-concept project idea, some sample code or just a link to your commits from other project)
- [x] I linked to my sample contribution from the proposal
- [x] I linked to my opened issues in numfocus/gsoc repository from the proposal

Mentors listed are:
1. @tacaswell 

I'm looking forward to working with the community and getting my proposal reviewed.
#201

### CODE REVIEW

File: 2017/proposals/satija-vidur.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+* This will require developing a significant new set of classes for Matplotlib, possibly parallel to possibly extending the existing 1D normalization and color mapping classes. 
+* These classes will need to be exposed to users either as new API or extending the existing `ScalerMappable` API.
+* In addition, a 2D color bar will be implemented and suitable color maps will be developed. 
+* Presumably no extra 3rd party libraries will be used except numpy.
+
+## Schedule of Deliverables
+
+### May 1th - May 28th, **Community Bonding Period**
+
+* Read the documentation, getting familiarized with the coding style and the community, version controlling, tests used and various systems the code will be deployed to. This includes going through the core documentation and getting to know how the code works by trying hands-on.
```
Comment (story645) [inline 2017/proposals/satija-vidur.md]: Anything specific here?

File: 2017/proposals/satija-vidur.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+Link : https://github.com/vidursatija/voiceai
+
+## Why this project?
+
+* It interests me. Plotting applications amaze me and I always wanted to work on one since I first used a plotting application online. I would also like to explore more in open source and develop a stronger grip on python and Machine Learning in general.
+
+* Matplotlib is an extensively used library for plotting. I've previously worked on matplotlib related to Colormaps. I found the code flexible and I loved working on it and would like to contribute more to the community.
+
+* As I already have a fairly solid idea of what to expect and the workings of matplotlib I believe I can contribute to this project in a substantial manner.
```
Comment (story645) [inline 2017/proposals/satija-vidur.md]: Can you please update your timeline to reflect this? Currently, it seems really thin on details.

File: 2017/proposals/satija-vidur.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+### June 12th - June 23th, **End of Phase 1**
+
+* Complete the rough classes developed.
+* Making programs to try the implementation.
+* Fixing bugs.
+
+### June 26 - July 7th, **Begin of Phase 2**
+
+* Finishing the creation of 2D Colormaps.
+* Adding exception handling.
```
Comment (story645) [inline 2017/proposals/satija-vidur.md]: I think exception handling should be baked into implementation?

**Comment by story645** [comment]
Please remember to submit your proposal to google (through the gsoc webpage) tonight! https://summerofcode.withgoogle.com/dashboard/

File: 2017/proposals/satija-vidur.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+* As I already have a fairly solid idea of what to expect and the workings of matplotlib I believe I can contribute to this project in a substantial manner.
+
+## Appendix
+
+#### Availability
+
+I will frequently visit the dentist in May and June, but it won't take up much time.
+
+On 22nd July, my college will reopen, and fortunately this date lies between the end of phase 2 and beginning of phase 3, so it won't affect the development.
```
Comment (tacaswell) [inline 2017/proposals/satija-vidur.md]: Will you be in school in August?

**Comment by kain88-de** [comment]
This application isn't valid. There hasn't been a final form submitted to the GSoC website for it.

**Pull request closed without merging**